### PR TITLE
Add support for read concern

### LINKED ISSUE #625: mongoengine compatibility with new version 0.20.0
Similar to issue #406 the new version of mongoengine stopped working with mongomock. Likely cause is the missing implementation of the function set_read_write_concern.

Sample code producing the TypeError:
```python
from mongoengine import connect, Document, fields

connect('mongoenginetest', host='mongomock://localhost')

class User(Document):
    _id = fields.IntField(primary_key=True)
    name = fields.StringField()


def print_all(collection):
    for obj in collection:
        print(obj.name)


user = User(_id=1, name="Max")
User.objects.insert(user)
print_all(User.objects)

User.objects(_id=1).update_one(name="Moritz")
print_all(User.objects)
```

With the output:
```
Max
Traceback (most recent call last):
[...]
File "venv/lib/python3.8/site-packages/mongoengine/context_managers.py", line 270, in set_read_write_concern
    combined_read_concerns = dict(collection.read_concern.document.items())
TypeError: 'Collection' object is not callable
```

### PR DESCRIPTION
**Pull request by abarto**
Adds support for setting the read concern at the collection level restoring compatibility with mongoengine 0.20.

Fixes: https://github.com/mongomock/mongomock/issues/625

### CODE REVIEW

File: mongomock/read_concern.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @property
+    def ok_for_legacy(self):
+        return True
+
+    @property
+    def document(self):
+        return self._document.copy()
+
+    def __eq__(self, other):
+        if isinstance(other, ReadConcern):
```
Comment (pcorpet) [inline mongomock/read_concern.py]: Please use an equality that would work with the pymongo ReadConcern (see `write_concern` for an example)

File: tests/test__collection_api.py [hunk 2]
```diff
@@ -1848,6 +1851,20 @@ def test__with_options_different_write_concern(self):
         self.assertNotEqual(self.db.collection.write_concern, col2.write_concern)
         self.assertEqual({'w': 2}, col2.write_concern.document)
 
+    def test__with_options_different_write_readconcern(self):
```
Comment (pcorpet) [inline tests/test__collection_api.py]: Please fix the name `...different_read_concern`

File: tests/test__collection_api.py [hunk 3]
```diff
@@ -1848,6 +1851,20 @@ def test__with_options_different_write_concern(self):
...
+        self.db.collection.insert_one({'name': 'col1'})
+        col2 = self.db.collection.with_options(read_concern=ReadConcern(level='majority'))
+        col2.insert_one({'name': 'col2'})
+
+        # Check that the two objects have the same data.
+        self.assertEqual({'col1', 'col2'}, {d['name'] for d in self.db.collection.find()})
+        self.assertEqual({'col1', 'col2'}, {d['name'] for d in col2.find()})
+
+        # Check that each object has its own write concern.
```
Comment (pcorpet) [inline tests/test__collection_api.py]: Update the comment here as well

File: tests/test__collection_api.py [hunk 4]
```diff
@@ -31,6 +31,7 @@
     from mongomock.collection import ReturnDocument
     from mongomock import ObjectId
     from mongomock.write_concern import WriteConcern
+    from mongomock.read_concern import ReadConcern
```
Comment (pcorpet) [inline tests/test__collection_api.py:34]: Please import the real ReadConcern in the other part of the branch

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. Please address my comments and try to fix the CI

**Review by pcorpet — APPROVED** [review]

**Comment by lsmar** [comment]
Any plans to integrate this into a release? 3.19.1?

**Pull request merged**